// gocardless-nodejs/services/balanceService.js
'use strict';
Object.defineProperty(exports, "__esModule", { value: true });
exports.BalanceService = void 0;
class BalanceService {
    constructor(api) {
        this.api = api;
    }
    async list(requestParameters) {
        const urlParameters = [];
        const requestParams = {
            path: '/balances',
            method: 'get',
            urlParameters,
            requestParameters,
            payloadKey: null,
            fetch: null,
        };
        const response = await this.api.request(requestParams);
        const formattedResponse = {
            ...response.body,
            __response__: response.__response__,
        };
        return formattedResponse;
    }
    async *all(requestParameters) {
        let cursor = undefined;
        do {
            const list = await this.list({ ...requestParameters, after: cursor });
            for (const balance of list.balances) {
                yield balance;
            }
            cursor = list.meta.cursors.after;
        } while (cursor);
    }
}
exports.BalanceService = BalanceService;
